ext/zlib/zlib.c

```diff
@@ -1032,8 +1032,8 @@ PHP_FUNCTION(inflate_get_status)
 	}
 
 	if (!(ctx = zend_fetch_resource_ex(res, NULL, le_inflate))) {
-		php_error_docref(NULL, E_WARNING, "Invalid zlib.inflate resource");
-		RETURN_FALSE;
+		zend_type_error("Invalid zlib.inflate resource");
```

Comment:
Here and below, I'd again suggest passing a name to zend_fetch_resource instead.